kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/KafkaConnectService.java

```diff
@@ -0,0 +1,228 @@
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<ConnectorStatus> getConnectorStatus(String clusterName, String connectorName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.get()
+                        .uri(cluster.getKafkaConnect() + "/connectors/{connectorName}/status", connectorName)
+                        .retrieve()
+                        .bodyToMono(ConnectorStatus.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error))
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<Void> restartConnector(String clusterName, String connectorName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.post()
+                        .uri(cluster.getKafkaConnect() + "/connectors/{connectorName}/restart", connectorName)
+                        .retrieve()
+                        .bodyToMono(Void.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error))
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<Void> pauseConnector(String clusterName, String connectorName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.put()
+                        .uri(cluster.getKafkaConnect() + "/connectors/{connectorName}/pause", connectorName)
+                        .retrieve()
+                        .bodyToMono(Void.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
+                        .doOnError(log::error))
+                .orElse(Mono.error(new NotFoundException("No such cluster")));
+    }
+
+    public Mono<Void> resumeConnector(String clusterName, String connectorName) {
+        return clustersStorage.getClusterByName(clusterName)
+                .map(cluster -> webClient.put()
+                        .uri(cluster.getKafkaConnect() + "/connectors/{connectorName}/resume", connectorName)
+                        .retrieve()
+                        .bodyToMono(Void.class)
+                        .retryWhen(
+                                Retry.onlyIf(e -> e.exception() instanceof WebClientResponseException.Conflict)
+                                        .retryMax(MAX_RETRIES))
```

Comment:
Let's move this duplicate retry logic to a method/variable

Reply:
Fixed, thanks